test/e2e/framework/ingress_utils.go

```diff
@@ -865,13 +865,19 @@ func (cont *GCEIngressController) GetFirewallRuleName() string {
 
 // GetFirewallRule returns the firewall used by the GCEIngressController.
 func (cont *GCEIngressController) GetFirewallRule() *compute.Firewall {
-	gceCloud := cont.Cloud.Provider.(*gcecloud.GCECloud)
-	fwName := cont.GetFirewallRuleName()
-	fw, err := gceCloud.GetFirewall(fwName)
+	fw, err := cont.GetFirewallRuleOrError()
 	Expect(err).NotTo(HaveOccurred())
 	return fw
 }
 
+// GetFirewallRule returns the firewall used by the GCEIngressController.
+// Returns an error if that fails.
+func (cont *GCEIngressController) GetFirewallRuleOrError() (*compute.Firewall, error) {
```

Comment:
We don't generally follow this pattern in k8s. The methods that return errors are called `GetFirewallRule()` because pretty much everything returns an error. The ones that fail are called `GetFirewallRuleOrDie()` or `GetFirewallRuleOrFail()`. Is it not appropriate to use that pattern here?

Reply:
It is but that would require updating all the methods here. If you search for `Expect(err).NotTo(HaveOccurred())` in this file, you will see that all Get/List methods are using it.

I can break that pattern for GetFirewallRule method only. That will make it consistent with rest of k/k code but inconsistent with all other methods in this file, which I think is worst.

I have added a TODO for now to rename all methods.